massgov/openmass
# DP-17625 Visual story field for layout sidebar display

**Description:**
This PR is for adding a new Page template dropdown field to the info_details content-type. When "Infomation Details" is selected from the dropdown nothing happens as this is the default. When "Visual Story" is selected the sidebar is not rendered and the main content fills the width of the screen. A new permission has been created 'set info details page template' that allows this to be displayed for users with certain roles. Only the `developer` and `content_team` roles currently have this permission checked. 

Please also see the PR in the mayflower repo:
https://github.com/massgov/mayflower/pull/969

**Jira:**
https://jira.mass.gov/browse/DP-17625

**To Test:**

1. Login as a user that has the `content_team` or `developer` roles enabled
2. Visit any Information Details page that has a sidebar.
3. Verify the sidebar appears and looks the same as it always has.
4. Edit the page and see the new dropdown field. It should default to "Information Details"
5. Select "Visual Story" and save the form
6. Now the page should not have the sidebar and the main content is full width
7. Verify this looks good on all screen sizes
8. Now login as a user without `content_team` or `developer` or super admin roles
9. Visit a different Information Details page and edit the page
10. Verify the new dropdown field does not display
11. Verify you can save the page without validation error


**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)
